client/trino-jdbc/src/test/java/io/trino/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -1149,13 +1149,15 @@ private void assertMaxRowsResult(Statement statement, long expectedCount)
         }
     }
 
-    @Test(expectedExceptions = SQLException.class, expectedExceptionsMessageRegExp = "Max rows exceeds limit of 2147483647")
+    @Test
     public void testMaxRowsExceedsLimit()
             throws Exception
     {
         try (ConnectedStatement connectedStatement = newStatement()) {
             connectedStatement.getStatement().setLargeMaxRows(Integer.MAX_VALUE * 10L);
-            connectedStatement.getStatement().getMaxRows();
+            assertThatThrownBy(() -> connectedStatement.getStatement().getMaxRows())
```

Comment:
`connectedStatement.getStatement()::getMaxRows`

-- you want to assert that getMaxRows() fails and getStatement() succeeds

Reply:
https://github.com/trinodb/trino/pull/21536